src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1894,42 +1977,81 @@ ApiResultWrapper addMemberToGroup(String group_id, String member_email, String m
+
+					if (statusCode / 100 != 2)  {
+						errorHandlingWhenNot200(statusObj, statusCode, ggbMsg);
+						log.error(String.format("Failure in migrating message with MessageId: \"%1$s\" to google groups" +
+								", status code %2$d and error message %3$s", messageId, statusCode, ggbMsg));
+						return new AsyncResult<String>(statusObj.toString());
+					}
 
-				String emailText = formatter.rfc822Format();
-
-				// mark the file as being processed
-				mRepository.setMigrationMessageStartTime(message.getMessage_id(), new Timestamp(System.currentTimeMillis()));
-
-				// process the message
-				ApiResultWrapper arw = addEmailToGoogleGroup(googleGroupId, emailText);
-				status = arw.getStatus().toString();
-				log.debug("uploadMessageToGoogleGroup: status: googleGroupId: {}",status,googleGroupId);
+					// Upload to Google groups went fine
+					String messageStatus = (String) statusObj.get(Utils.JSON_ATTR_ITEM_STATUS);
+					String messageStr = (String) statusObj.get(Utils.JSON_ATTR_MESSAGE);
+					//This is the case when in the EmailFormatter attachment might have dropped due to some error or size limit
+					if (messageStatus == Utils.STATUS_PARTIAL) {
+						statusObj.put(Utils.JSON_ATTR_ITEM_STATUS, "Google Groups upload Went fine but " + messageStr);
+					}
+					log.info("The response from google groups when 200: "+ggbResult+" for MessageId: "+messageId);
+				}
 
-			}
-			catch (IOException exception)
-			{
+			}catch (IOException exception) {
 				String errorString = "IOException from EmailFormatter for message id " + messageId + " " + exception.getMessage();
 				log.error(errorString);
-				status = errorString;
-			}
-			catch (Exception e) {
+				errHandlingWhenExceptions(statusObj);
+			}catch (ParseException e){
+				String errorString = "ParseException while extracting response from GGB with message id " + messageId + " " + e.getMessage();
+				log.error(errorString);
+				errHandlingWhenExceptions(statusObj);
+			}catch (Exception e) {
 				String msg = String.format("unexpected exception in uploadMessageToGoogleGroup: %s for messageId: %s",
 						e.getMessage(),messageId);
 				log.error(msg);
-				status = msg;
+				errHandlingWhenExceptions(statusObj);
 			}
+
 			finally {
 				// update the status and end time for file item
 				mRepository.setMigrationMessageEndTime(messageId, new java.sql.Timestamp(System.currentTimeMillis()));
-				mRepository.setMigrationMessageStatus(messageId, status);
+				mRepository.setMigrationMessageStatus(messageId, statusObj.toString());
 			}
-
 			log.debug("uploadMessageToGoogleGroup: return");
-			return new AsyncResult<String>(status);
+			return new AsyncResult<String>(statusObj.toString());
 		}
 
+	private void errHandlingWhenExceptions(JSONObject statusObj) {
+		statusObj.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_ERROR);
+		statusObj.put(Utils.JSON_ATTR_MESSAGE, "Failure in upload to Google Groups");
```

Comment:
Yes, dave when there is any sort of exception or error cases i am logging it. If you look at the exception there you will find log.error statement and a method call as above to put that state in the Report object